articles/http-requests-nodejs/index.md

```diff
@@ -0,0 +1,167 @@
+
+### Method and Description
+1. GET: GET method is used to establish connections and receive info from the server. It is used while making API calls, where no modification of data is involved.
+2. POST: The POST method is generally used to send data inside the entity-body section. Authentication, File Uploads, etc. are all done via POST method requests.
+3. DELETE: This method is used to delete resources on the server.
+4. PUT: This method is used to replace existing resources on the server with the updated resources.
+5. HEAD: Functions similarly to GET, but sends the message request without the entity-body. This is used in cases when the server must not return the message-body in the response. Or in cases when the files being accessed are too large to be transmitted.
+6. OPTIONS: This method is used when the client wants to understand the various communication methods that the server supports.
+7. TRACE: Used for testing purposes. The message is sent from the client to the server, and the route is logged.
+
+### Libraries in JS
+1. HTTP
+2. fetch
+3. Axios
+4. node-fetch
+5. got
+
+### Code implementation
+#### http module
+The HTTP module is available with Node.js. There is no additional installation required. The data is initially converted into a string using the stringify function. The HTTP options specify the headers, destination address, and request method type. Next, we use http.request to send the data to the server and await the response. The response is stored in the req variable, and upon error, it is logged into the console. On successful transmission, the data is posted to the server.
+
+```jsx
+
+const data = querystring.stringify({
+  'msg': 'Hello World!'
+});
+
+const options = {
+        hostname: '127.0.0.1',
+        port: 5000,
+        path: '/',
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+            'Content-Length': data.length,
+            'Access-Control-Allow-Origin': '*'
+        },
+    }
+
+const req = http.request(options, (res) => {
+        console.log(`HEADERS: ${JSON.stringify(res.headers)}`);
+        console.log(`statusCode: ${res.statusCode}`)
+        res.on('data', (chunk) => {
+          console.log(`BODY: ${chunk}`);
+        });
+        res.on('end', () => {
+          console.log('No more data in response.');
+        });
+    })
+
+req.on('error', (error) => {
+        console.error(error)
+    })
+
+req.write(data)
+req.end()
+```
+
+### fetch
+The fetch module is also built-in and is usually used for making API calls. Fetch predominantly uses the concept of promises. Promises are objects that represent a point in time when the completion or failure of an asynchronous task occurs. Errors in any then statements is caught by the catch statement. Therefore, error handling is much more simplified.
+
+
+```jsx
```

Comment:
@lalith1403 The `fetch` code block needs some attention. Looks like the code fencing got a little jumbled, and there's some axios code mixed in there as well.

Reply:
yikes... I was testing axios code out and had pasted it there for reference. sorry... I am removing those lines right away... the ones that havent been included in the code blocks were for my reference